infrastructure/api_management.tf

```diff
@@ -17,6 +17,42 @@ module "api-management" {
   virtual_network_configuration = [module.subnets_hub["${module.config[each.key].names.subnet}-api-mgmt"].id]
   zones                         = var.apim_config.zones
 
+  developer_portal_hostname_configuration = [
+    for domain in var.apim_config.custom_domains : {
```

Comment:
you could improve this by putting it into a single loop rather than having 4 loops doing the same thing.